File: public/javascripts/directives/CountryDirective.js
(function() {
    var app = angular.module('shocktrade');

    /**
     * Country Directive
     * @author lawrence.daniels@gmail.com
     * <country id="{{ p.facebookID }}" style="width: 24px; height: 24px"/>
     */
    app.directive('country', ['$log', function($log) {
        return {
            restrict: 'E',
            scope: { profile:'@profile' },
            transclude: true,
            replace: false,
            template: '<img ng-src="{{ url }}" style="vertical-align: middle" />',
            link: function(scope, element, attrs) {
                scope.url = "/assets/images/country/us.png";
            }
        };
    }]);

})();

File: app/client/angularjs/src/main/scala/com/shocktrade/client/directives/CountryDirective.scala
package com.shocktrade.client.directives

import io.scalajs.npm.angularjs.Directive.{ElementRestriction, LinkSupport, TemplateSupport}
import io.scalajs.npm.angularjs.{Attributes, Directive, JQLite, Scope}

import scala.scalajs.js
import scala.scalajs.js.annotation.ScalaJSDefined

/**
  * Country Directive
  * @author Lawrence Daniels <lawrence.daniels@gmail.com>
  */
class CountryDirective extends Directive with ElementRestriction with LinkSupport[CountryDirectiveScope] with TemplateSupport {

  override def scope = new CountryDirectiveInputs(profile = "@profile")

  override def template = """<img ng-src="{{ url }}" style="vertical-align: middle" />"""

  override def link(scope: CountryDirectiveScope, element: JQLite, attrs: Attributes): Unit = {
    scope.$watch("profile", (newProfile: js.Any, oldProfile: js.Any) => {
      scope.url = "/images/country/us.png"
    })
  }

}

/**
  * Country Directive Input Parameters
  * @author Lawrence Daniels <lawrence.daniels@gmail.com>
  */
@ScalaJSDefined
class CountryDirectiveInputs(val profile: String) extends js.Object

/**
  * Country Directive Scope
  * @author Lawrence Daniels <lawrence.daniels@gmail.com>
  */
@js.native
trait CountryDirectiveScope extends CountryDirectiveInputs with Scope {
  // output fields
  var url: js.UndefOr[String] = js.native
}
